test/index.test.ts

```diff
@@ -416,6 +417,62 @@ describe('run', function () {
     })
   }) // end 'target'
 
+  describe('distTag as target', () => {
+    /** Compact function to test ncu */
+    const testNcuTargetUpgrade = async (target: Target, currentVersion: string, upgradeVersion: string) => {
+      const packageName = 'ncu-test-tag'
+      const pkgData = await ncu.run({
+        target,
+        packageData: `{ "dependencies": { "${packageName}": "${currentVersion}" } }`,
+      })
+
+      if (currentVersion === upgradeVersion) {
+        pkgData!.should.not.have.property(packageName)
+      } else {
+        // A conflict between eslint and prettier up ahead
+        // eslint-disable-next-line @typescript-eslint/no-extra-semi
+        ;(pkgData as any)[packageName].should.equal(upgradeVersion)
+      }
+    }
+
+    it('latest <--> empty dist-tag', async () => {
+      // FIXME: uncomment once 1.0.0-1 and 1.0.0.0 are uploaded
+      // await testNcuTargetUpgrade('@next', '0.1.0', '1.0.0-1')
+      // await testNcuTargetUpgrade('@next', '0.1.0', '1.0.0-1')
+      //
+      // await testNcuTargetUpgrade('@next', '0.1.0', '1.0.0-1')
+      // await testNcuTargetUpgrade('@next', '0.2.0-0', '1.0.0-1')
+      await testNcuTargetUpgrade('latest', '1.0.0-1', '1.1.0')
+    })
+
+    it('upgrade to higher semver between dist-tags', async () => {
+      await testNcuTargetUpgrade('@beta', '1.0.0-task-42.0', '1.0.1-beta.0')
+    })
+
+    it('upgrade to same semver between dist-tags', async () => {
+      // can't detect which prerelease is higher, so just allow switching
+      await testNcuTargetUpgrade('@task-42', '1.0.0-beta.0', '1.0.0-task-42.0')
+
+      // same here, major/minor/patch is the same, can't determine which one is bigger task-42 or next
+      // await testNcuTargetUpgrade('@next', '1.0.0-task-42.0', '1.0.0-1')
+      await testNcuTargetUpgrade('@next', '1.0.0-task-42.0', '0.2.0-0')
```

Comment:
Easiest way to explain why is this:
1. Latest is at 0.1.0
2. First developer does npm version premajor --preid=task-42 && npm publish --tag=task-42
3. Second developer goes npm version preminor && npm publish --tag=next
4. Since both branches are made out of 0.1.0, can we really say that 1.0.0-task-42 is greater than 0.2.0-0 ? They are both prereleases, so there was no release
There is no way to "fix" this for second-dev Because merging task-42 into next is something that should not be done as task-42 could be in a separate testing stage. So I suggest we use the intention, if somebody wants to go from one prerelease branch to another - let them.
Normally all the upgrading should be done from latest -> dist-tag -> latest, or within same dist-tag.